### ISSUE TITLE
Flow asymmetry. 

### ISSUE THREAD

**Issue by pociej** [nicolodavis/boardgame.io#308]
Look at the following example please, this is kinda pseudocode but hope enough to show the clue of the issue.  
```
const phases = {
  phase_1: {
    endPhaseIf: G => {
      console.log("e_1");
    },
    onMove: G => {
      console.log("m_2");
      return G;
    },
    next: "phase_2"
  },
  phase_2: {
    endPhaseIf: G => {
      console.log("e_2");
    },
    onMove: G => {
      console.log("m_2");
      return G;
    },
    next: "phase_1"
  }
};

const game = new Game({
  ...setup,
  flow: {
    phases,
    startingPhase: "phase_1"
  }
});

```

When you run game with such setup what you will see in console will be in order : 
`m_1`, `e_1`, `e_2`, `m_2`, `e_2`, `e_1` .... 
What i mean is that at the beginning of the game, `endPhaseIf` for starting phase is not called, but each subsequent phase starts with `endPhaseIf`. Its not end of the world, but imo is not elegant.

**Comment by Stefan-Hanke** [comment]
Agreed, it's not needed because the game could just as well start with another phase, but it should work the same. This hints at duplicated code somewhere, or maybe just an omission.

**Comment by nicolodavis** [comment]
I'm actually having trouble parsing this.

@pociej I'm assuming that `m_2` in `phase_1` is actually `m_1`. Are both `endPhaseIf` functions meant to return `true` always?

**Comment by nicolodavis** [comment]
> What endPhaseIf returns does not matter

It does because your phase will (or will not) end depending on the return value of this function. I understand that you may have omitted details from your pseudocode, but I'm trying to understand if you intended for it to always return `true` (i.e. how are these phases actually ending?).  Also, I'm assuming that your console output is after the result of a single move being played in Phase 1?

**Comment by pociej** [comment]
@nicolodavis Lets assume then, that `endGameIf` returns true when there was at least one more performed in given phase to give `onMove` chance to fire.

**Comment by nicolodavis** [comment]
Oh I see.

So your intended sequence of events is:
1. Play a move in Phase 1
2. Phase 1 ends
3. Play a move in Phase 2
4. Phase 2 ends

Correct?

**Comment by pociej** [comment]
Something like this or similar. Its not like that i cant do something. I just think that asymmetry is bad. Now endPhaseIf is called on start of each phase except starting which is a bit strange.

**Comment by nicolodavis** [comment]
This should be fixed in v0.33.  Feel free to re-open if you still have problems with this.
